src/lib/stores.ts

```diff
@@ -0,0 +1,130 @@
+import { writable, derived } from 'svelte/store';
+import { withoutTransition } from './without-transition';
+
+// saves having to branch for server vs client
+const noopStorage = {
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	getItem: (_key: string) => null,
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	setItem: (_key: string, _value: string) => {}
+};
+
+/**
+ * The key used to store the mode in localStorage.
+ */
+export const localStorageKey = 'mode';
+/**
+ * Writable store that represents the user's preferred mode (`"dark"`, `"light"` or `"system"`)
+ */
+export const userPrefersMode = createUserPrefersMode();
+/**
+ * Readable store that represents the system's preferred mode (`"dark"`, `"light"` or `undefined`)
+ */
+export const systemPrefersMode = createSystemMode();
+/**
+ * Derived store that represents the current mode (`"dark"`, `"light"` or `undefined`)
+ */
+export const derivedMode = createDerivedMode();
+
+// derived from: https://github.com/CaptainCodeman/svelte-web-storage
+function createUserPrefersMode() {
+	const browser = typeof window !== 'undefined' && typeof document !== 'undefined';
```

Comment:
We make this just `typeof document !== 'undefined'`  as certain runtimes can define `window`, such as Deno.

See: https://remix.run/docs/en/1.16.1/pages/gotchas#typeof-window-checks for reference.

```suggestion
	const isBrowser = typeof document !== 'undefined';
```

Sorry for the name nitpicking, just reads better if something can only be a boolean prefixed by `is` 😃

Reply:
thanks for the heads up and the name suggestion, I just quickly refactored from using `esm-env` and it's `BROWSER` variable so did not think much of it. will fix!